File: AizuOnline/ALDS/ALDS1_4_B_Binary Search.cpp
#include <iostream>

using namespace std;

bool binarySearch(int* array,int n,int value){
	int left=0;
	//鉴于整数除法向下取整，可以给right一个永远不会取到的值，成为border
	int right=n;
	//以便把小于等于换成小于，左闭右开
	while(left<right){
		int mid=(left+right)/2;
		if(value==array[mid]) return true;
		//left照常
		if(value>array[mid]) left=mid+1;
		else right=mid;
	}
	return false;
}

int main(){
	int n;
	int S[100000];
	int q,T;
	int C=0;
	cin>>n;
	for(int i=0;i<n;i++) cin>>S[i];
	cin>>q;
	for(int i=0;i<q;i++) {
		cin>>T;
		if(binarySearch(S,n,T)) C++;
	}
	cout<<C<<endl; 
}

File: AizuOnline/Review_Py/ALDS1_4_B_Binary Search.py
N = int(input())
S = list(map(int, input().split()))
Q = int(input())
T = list(map(int, input().split()))

def binary_search(lis, x) -> int:
    left, right = 0, len(lis)
    while left < right:
        mid = (left + right) // 2
        if lis[mid] == x:
            return True
        elif lis[mid] < x:
            left = mid + 1
        elif lis[mid] > x:
            right = mid
    return False

cnt = 0
for x in T:
    cnt += binary_search(S, x)

print(cnt)

